src/navigator.cpp

```diff
@@ -0,0 +1,58 @@
+#include "navigator.h"
+#include "Graph.h"
+
+// File: navigator.cpp
+// Purpose: Class implementing Dijkstra's Algorithm for finding the shortest path in a graph.
+
+
+class Navigator : Graph {
```

Comment:
I like this idea of having a `Navigator` class that is used for, well, any sort of navigation through out a `Graph` object. 

I was thinking that it would be good to set up either a `class` or `namespace` for other algorithms that may be needed in the future. `Navigator` could be just one of several, though it should posses the algorithms related to shortest/longest path, existence of path, etc. Just a thought.

Reply:
I was thinking of algorithms like whether or not a graph is connected to start with. Just so we have simple tools to determine if a graph is what we consider to be correct.